src/sql/workbench/contrib/notebook/browser/cellViews/linkHandler.directive.ts

```diff
@@ -38,19 +41,13 @@ export class LinkHandlerDirective {
 				return;
 			}
 		}
-		try {
-			const href = target['href'];
-			if (href) {
-				this.handleLink(href);
-			}
-		} catch (err) {
-			onUnexpectedError(err);
-		} finally {
-			event.preventDefault();
+		const href = target['href'];
+		if (href) {
+			await this.handleLink(href).catch(e => onUnexpectedError(e)).finally(event.preventDefault);
```

Comment:
So if you're removing the try/catch and adding the .catch then this function doesn't need to be async. This will still work fine as is - it's just unnecessary though to make it async.

We generally prefer using async functions though so I'd suggest getting rid of the .catch and instead adding the try/catch/finally back

Reply:
I made the method async and removed the try/catch from the promise 👍